### PR TITLE
#MM-12130 - Fix incorrect key in en.json and changes some translations from service terms to terms of service

### PR DESCRIPTION
**Pull request by harshilsharma**
#### Summary
Fix incorrect key in en.json and changes some translations from service terms to terms of service

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12130

#### Checklist
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -4752,7 +4752,7 @@
   },
   {
     "id": "model.service_terms.is_valid.text.app_error",
-    "translation": "Invalid terms of service text."
+    "translation": "Terms of service text is too long."
```
Comment (jasonblais) [inline i18n/en.json]: Propose `Custom terms of service text is too long. Maximum {MaxLength} characters allowed`

where `{MaxLength}` is the max character size, I think around 16,000?

File: i18n/en.json [hunk 2]
```diff
@@ -6384,19 +6384,19 @@
...
   },
   {
     "id": "store.sql_service_terms_store.get.app_error",
-    "translation": "Unable to fetch service terms"
+    "translation": "Unable to fetch terms of service"
   },
   {
     "id": "store.sql_service_terms_store.get.no_rows.app_error",
-    "translation": "No service terms found"
+    "translation": "No terms of service found"
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply i18n/en.json]: Done.

File: i18n/en.json [hunk 3]
```diff
@@ -6384,19 +6384,19 @@
...
   },
   {
     "id": "store.sql_service_terms.save.app_error",
-    "translation": "Unable to save service terms"
+    "translation": "Unable to save terms of service"
   },
   {
     "id": "store.sql_service_terms_store.get.app_error",
-    "translation": "Unable to fetch service terms"
+    "translation": "Unable to fetch terms of service"
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply i18n/en.json]: Done,

File: i18n/en.json [hunk 4]
```diff
@@ -6384,19 +6384,19 @@
   },
   {
     "id": "store.sql_service_terms_store.save.existing.app_error",
-    "translation": "Must not call save for existing service terms"
+    "translation": "Must not call save for existing terms of service"
   },
   {
     "id": "store.sql_service_terms.save.app_error",
-    "translation": "Unable to save service terms"
+    "translation": "Unable to save terms of service"
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply i18n/en.json]: Done,

File: i18n/en.json [hunk 5]
```diff
@@ -6384,19 +6384,19 @@
   },
   {
     "id": "store.sql_service_terms_store.save.existing.app_error",
-    "translation": "Must not call save for existing service terms"
+    "translation": "Must not call save for existing terms of service"
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply i18n/en.json]: Done.

File: i18n/en.json [hunk 6]
```diff
@@ -3216,7 +3216,7 @@
   },
   {
     "id": "ent.cluster.save_config.update_custom_service_terms_no_user.error",
-    "translation": "Custom service terms can only be changed if provided with exactly one user id"
+    "translation": "Custom terms of service can only be changed if provided with exactly one user id"
```
[the remark this answers is not in the document]
Comment (harshilsharma) [reply i18n/en.json]: Yes, I mean user id. Actually, this message is not being used now. Hence, removing it.

File: i18n/en.json (hunk 1 above)
Comment (harshilsharma) [reply i18n/en.json]: Done. It's 16384 characters.

**Comment by jasonblais** [comment]
@cpanato ready to merge and cherry-pick for v5.4

**Comment by cpanato** [comment]
@jasonblais 
done cherry pick (71e3afaf63631583202f824424ced2e3aa02d140) to release 5.4

**Pull request merged**